src/bun.js/bindings/webcore/JSEventListener.cpp

```diff
@@ -234,6 +251,32 @@ void JSEventListener::handleEvent(ScriptExecutionContext& scriptExecutionContext
     if (handleExceptionIfNeeded(uncaughtException))
         return;
 
+    // Node handles promises in the return value and throws an uncaught exception on nextTick if it rejects.
+    // See event_target.js function addCatch in node
+    if (retval.isObject()) {
+        auto then = retval.get(lexicalGlobalObject, vm.propertyNames->then);
```

Comment:
Can we check if it's a promise instead? Or does it work on arbitrary thenables?

Reply:
The node implementation checked for `then`